qiskit/user_config.py

```diff
@@ -60,21 +64,35 @@ def read_config_file(self):
                                           'latex_source', 'auto']:
                     raise exceptions.QiskitUserConfigError(
                         "%s is not a valid circuit drawer backend. Must be "
-                        "either 'text', 'mpl', 'latex', 'auto', or "
-                        "'latex_source'"
+                        "either 'text', 'mpl', 'latex', 'latex_source', or "
+                        "'auto'."
```

Comment:
what's `auto`?

Reply:
If 'auto', circuit_drawer checks _matplotlib.HAS_MATPLOTLIB. If true use 'mpl', else 'text'. Don't know if anybody uses it, but has been in code since before my time.

Reply:
Is it documented?

Reply:
Not currently. May have been in the past. Functionality is in circuit_drawer, but this is only doc reference.

Reply:
It was added in https://github.com/Qiskit/qiskit-terra/pull/2331 if I remember the use correctly it was for the case where people run qiskit with the same code some where mpl works others where it didn't. So setting auto in your user config enables you to call a bare draw() and have it use mpl if available else it would fallback to text.

It wasn't documented when it was added mostly because we didn't have a good place to document it back then, and the documentation was also in a state of flux back then (it was around the time of adding multiple elements). We probably should mention of it in: https://github.com/Qiskit/qiskit/blob/master/docs/configuration.rst

Reply:
That seems like a good place to document it. We need to add `circuit_mpl_style_path` to the list as well, so I can take care of both of those.

Reply:
Maybe also here https://qiskit.org/documentation/stubs/qiskit.visualization.circuit_drawer.html?highlight=Valid%20choices%20are ? It used to be default?

Reply:
Actually, I think this is only for the user_config based on @mtreinish first paragraph comment above. A user wouldn't use it if calling QuantumCircuit.draw or circuit_drawer.